lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,424 @@
+            "OBS_MODE": evt_hdr["OBS_MODE"],
+            "N_TELS": evt_hdr["N_TELS"],
+            "TELLIST": evt_hdr["TELLIST"],
+            "INSTRUME": evt_hdr["INSTRUME"],
+        }
+        obs_index_tables.append(t_obs)
+
+    obs_index_table = QTable(obs_index_tables)
+
+    obs_index_header = DEFAULT_HEADER.copy()
+    obs_index_header["HDUCLAS1"] = "INDEX"
+    obs_index_header["HDUCLAS2"] = "OBS"
+    obs_index_header["INSTRUME"] = t_obs["INSTRUME"]
+    obs_index_header["MJDREFI"] = evt_hdr["MJDREFI"]
+    obs_index_header["MJDREFF"] = evt_hdr["MJDREFF"]
+
+    obs_index = fits.BinTableHDU(
+        obs_index_table, header=obs_index_header, name="OBS INDEX"
+    )
+    obs_index_list = fits.HDUList([fits.PrimaryHDU(), obs_index])
+
+    return obs_index_list
+
+
+def create_hdu_index_hdu(
+    filename_list,
+    fits_dir,
+    hdu_index_filename,
+    add_fits_dir_meta=False,
+):
+    """
+    Create the hdu index table
+
+    A two-level index file scheme is used in the IACT community to allow
+    an arbitrary folder structures. For each directory tree, two files should be present:
+    1. obs-index.fits.gz
+    (http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html)
+    2. hdu-index.fits.gz
+    (http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html)
+
+    hdu-index contains the informations about the locations of the other
+    HDU (header data units) necessary to the analysis.
+
+    File directory information is currently kept optional in the HDU index table
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : Path
+        directory containing the fits file
+    hdu_index_filename : str
+        filename for HDU index file
+    add_fits_dir_meta : Bool
+        True for adding directory path of FITS file in HDU index table
+
+    Returns
+    -------
+    hdu_index_list : 'astropy.io.fits.HDUList'
+        HDU list file for HDU index tables
+    """
+
+    hdu_index_tables = []
+
+    # loop through the files
+    for file in filename_list:
+        filepath = fits_dir / file
+        if filepath.is_file():
+            try:
+                hdu_list = fits.open(filepath)
+                evt_hdr = hdu_list["EVENTS"].header
+                gti_hdr = hdu_list["GTI"].header
+                pnt_hdr = hdu_list["POINTING"].header
+            except Exception:
+                log.error(f"fits corrupted for file {file}")
+                continue
+        else:
+            log.error(f"fits {file} doesn't exist")
+            continue
+
+        # The column names for the table follows the scheme as shown in
+        # https://gamma-astro-data-formats.readthedocs.io/en/latest/general/HDU_CLASS.html
+        # Event list
+        t_events = {
+            "OBS_ID": evt_hdr["OBS_ID"],
+            "HDU_TYPE": "events",
+            "HDU_CLASS": "events",
+            "FILE_DIR": "",
```

Comment:
If I understand the docs correctly, `FILE_NAME` should only be the name and `FILE_DIR` the directory. E.g. to get the path, you would do `os.path.join(path_to_hdu_index, FILE_DIR, FILE_NAME)`.

Reply:
I have added the `BASE_DIR` information to make it more uniform. Now the path to the file would be `os.path.join(BASE_DIR, FILE_DIR, FILE_NAME)`. Gammapy also looks for the `BASE_DIR` for locating the HDUs.
I just had a small issue when running the test on this function alone as the current working directory was not the same. So I just made an extra parameter for this function. Can you suggest a cleaner way to do this?
`run_tools`, take care of the cwd value, so there was no problem in testing the Tool.